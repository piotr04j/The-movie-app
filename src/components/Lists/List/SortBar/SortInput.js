import React,{ Component } from 'react';

class SortInput extends Component {

    state = {
        value: 'descending'
    }

    handeValue = (e) => {
        this.setState({ value: e.target.value})
        this.props.getOptionValue(e.target.value)
    }

    render(){
        const {label, optionInputs} = this.props;
        return (
            <div className="col">
                <label className="sr-only"  htmlFor="inlineFormCustomSelectPref--one">{label}</label>
                <select className="custom-select"  id="inlineFormCustomSelectPref--one" onChange={(e) => this.handeValue(e)} value={this.state.value}>
                    {optionInputs.map( optionItem => {
                        return <option key={optionItem.val} value={optionItem.val}>{optionItem.content}</option>
                    })}
                </select>
            </div>
        )
    }
}

export default SortInput;
