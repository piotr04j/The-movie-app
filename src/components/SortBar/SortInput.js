import React from 'react';

const SortInput = ({label, optionInputs, getOptionValue}) => {

    return (
            <div className="col">
                <label className="sr-only"  htmlFor="inlineFormCustomSelectPref--one">{label}</label>
                <select className="custom-select"  id="inlineFormCustomSelectPref--one" onChange={(e) => getOptionValue(label,e.target.value)}>
                    {optionInputs.map( optionItem => {
                        return <option key={optionItem.val} value={optionItem.val}>{optionItem.content}</option>
                    })}
                </select>
            </div>
    )
}

export default SortInput;
