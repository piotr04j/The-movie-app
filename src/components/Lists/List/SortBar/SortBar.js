import React from 'react';
import SortInput from './SortInput';

const SortBar = ({handleSetOrder, sortValue}) => {

    return (
        <form className="form-inline container justify-content-end">
            <div className="form-row mt-3 ">
                <SortInput
                    optionInputs={[{content: 'Descending', val: 'descending'}, {content: 'Ascending', val: 'ascending'}]}
                    label="Order"
                    getOptionValue={handleSetOrder}
                />
            </div>
        </form>
    )
}

export default SortBar
