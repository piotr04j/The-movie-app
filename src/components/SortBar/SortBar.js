import React from 'react';
import SortInput from './SortInput';

const SortBar = ({getOptionValue}) => {
    return (
        <form className="form-inline container justify-content-end">
            <div className="form-row mt-3 ">
                <SortInput
                    optionInputs={[{content: 'Movies', val: 'movie'}, {content: 'TV Shows', val: 'tv'}]}
                    label="Check"
                    getOptionValue={getOptionValue}
                />
                <SortInput 
                    optionInputs={[{content: 'Top rated', val: 'vote_average'}, {content: 'Popular', val: 'popularity'}]}
                    label="Sort by"
                    getOptionValue={getOptionValue}
                />
                <SortInput
                    optionInputs={[{content: 'Descending', val: 'descending'}, {content: 'Ascending', val: 'ascending'}]}
                    label="Order"
                    getOptionValue={getOptionValue}
                />
            </div>
        </form>
    )
}

export default SortBar