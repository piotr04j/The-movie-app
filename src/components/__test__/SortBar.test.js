import React from 'react';
import renderer from 'react-test-renderer';
import SortBar from '../Lists/List/SortBar/SortBar';


it('renders correctly', () => {
    const tree = renderer.create(<SortBar/>).toJSON();
    expect(tree).toMatchSnapshot();
});