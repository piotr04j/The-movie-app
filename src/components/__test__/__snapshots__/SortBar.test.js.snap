// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  className="form-inline container justify-content-end"
>
  <div
    className="form-row mt-3 "
  >
    <div
      className="col"
    >
      <label
        className="sr-only"
        htmlFor="inlineFormCustomSelectPref--one"
      >
        Order
      </label>
      <select
        className="custom-select"
        id="inlineFormCustomSelectPref--one"
        onChange={[Function]}
        value="descending"
      >
        <option
          value="descending"
        >
          Descending
        </option>
        <option
          value="ascending"
        >
          Ascending
        </option>
      </select>
    </div>
  </div>
</form>
`;
